aten/src/ATen/native/native_functions.yaml

```diff
@@ -3847,6 +3847,27 @@
   dispatch:
     CompositeExplicitAutograd: slice
 
+- func: slice_scatter(Tensor self, Tensor mutated_view, int dim=0, int? start=None, int? end=None, int step=1) -> Tensor
```

Comment:
The arguments name `mutated_view` doesn't look right.
All our scatter ops use `src` for the Tensor being scattered.

Reply:
Well, they're not terribly consistent about it; `masked_scatter` takes `source` and `scatter` takes `src` except when it's a non-Tensor; then it's `value`. Agreed that a name like source is probably better though.

By the way, if we're talking about conventions, most of the pre-existing scatter functions take source at the very end, rather than the beginning. But I like Brian's new convention better as it handles optional arguments in the original view function better (as is the case in this example).

Reply:
Yes.
Also the name `mutated_view` is very specific to the way we want to use it in the context of the functionalization. But doesn't really make sense otherwise.

Reply:
I'll rename to `mutated_view` to `src`